cmd/antrea-agent/options.go

```diff
@@ -122,7 +122,10 @@ func (o *Options) validate(args []string) error {
 	if !ok {
 		return fmt.Errorf("TrafficEncapMode %s is unknown", o.config.TrafficEncapMode)
 	}
-
+	ok, _ = config.GetIPsecAuthenticationModeFromStr(o.config.IPsec.AuthenticationMode)
```

Comment:
You can also choose to add a field to Options to save the parsed mode, so no need to parse again. No strong preference. Up to you.